src/lockfile/wrapper.js

```diff
@@ -94,33 +96,36 @@ export default class Lockfile {
     [key: string]: LockManifest,
   };
 
+  parseResult: ?ParseResultType;
+
   static async fromDirectory(dir: string, reporter?: Reporter): Promise<Lockfile> {
     // read the manifest in this directory
     const lockfileLoc = path.join(dir, constants.LOCKFILE_FILENAME);
 
     let lockfile;
     let rawLockfile = '';
+    let parseResult;
 
     if (await fs.exists(lockfileLoc)) {
       rawLockfile = await fs.readFile(lockfileLoc);
-      const lockResult = parse(rawLockfile, lockfileLoc);
+      parseResult = parse(rawLockfile, lockfileLoc);
 
       if (reporter) {
-        if (lockResult.type === 'merge') {
+        if (parseResult.type === 'merge') {
           reporter.info(reporter.lang('lockfileMerged'));
-        } else if (lockResult.type === 'conflict') {
+        } else if (parseResult.type === 'conflict') {
           reporter.warn(reporter.lang('lockfileConflict'));
         }
       }
 
-      lockfile = lockResult.object;
+      lockfile = parseResult.object;
     } else {
       if (reporter) {
         reporter.info(reporter.lang('noLockfileFound'));
       }
     }
 
-    return new Lockfile(lockfile, rawLockfile);
+    return new Lockfile(lockfile, rawLockfile, parseResult && parseResult.type);
```

Comment:
I think a ternary would better convey the semantic, wdyt? The `&&` behavior obfuscates the fact that the return type is `parseResult.type`.